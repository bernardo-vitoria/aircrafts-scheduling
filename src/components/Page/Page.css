.page {
    padding-top: 32px;
    min-width: 900px;
}