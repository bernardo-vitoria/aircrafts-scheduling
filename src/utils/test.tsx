// @ts-nocheck
import React from "react";
import "@testing-library/jest-dom/extend-expect";
import { I18nextProvider } from "react-i18next";
import { QueryClient, QueryClientProvider } from "react-query";
import { MemoryRouter } from "react-router-dom"; // 4.0.0
import { queries } from "@testing-library/react";
import {
  render,
  RenderResult,
} from "@testing-library/react";
import i18n from "../config/i18nForTests";

const customRender = (ui: React.ReactElement, options = {}): RenderResult => {
  const CombinedProviders: React.FC = ({ children }) => {
    const config = {
      queries: {
        retry: false,
        cacheTime: 0,
        refetchOnWindowFocus: false,
        retryDelay: 0,
      },
    };

    const queryClient = new QueryClient({ defaultOptions: config });

    return (
        <QueryClientProvider client={queryClient}>
          <I18nextProvider i18n={i18n}>
            <MemoryRouter>{children}</MemoryRouter>
          </I18nextProvider>
        </QueryClientProvider>
    );
  };

  return render(ui, { wrapper: CombinedProviders, queries, ...options });
};

export {
  customRender as render
};