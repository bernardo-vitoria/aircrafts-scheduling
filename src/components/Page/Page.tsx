import * as React from "react";
import Container from "@mui/material/Container";
import "./Page.css";

const Page: React.FC = ({children}) => {
  return (
    <div className="page">
      <Container>{children}</Container>
    </div>
  );
};

export default Page;
