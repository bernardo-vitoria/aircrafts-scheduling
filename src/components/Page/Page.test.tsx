import Page from "./Page";
import { render } from "../../utils/test";

describe("Page component", () => {
  it("should match snapshot", () => {
    const { container } = render(<Page />);

    expect(container).toMatchSnapshot();
  });
});
