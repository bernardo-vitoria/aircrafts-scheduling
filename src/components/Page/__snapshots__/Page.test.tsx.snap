// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page component should match snapshot 1`] = `
<div>
  <div
    class="page"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    />
  </div>
</div>
`;
